### PR TITLE
Add support for azure-core credential in CLI factory

### LINKED ISSUE #7035: Support for get_client_from_cli_profile(keyClient)
is it possible to use get_client_from_cli_profile(keyClient) ?

In [4]: from azure.common.client_factory import get_client_from_cli_profile

In [5]: c = get_client_from_cli_profile(KeyClient)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-cee3faeda852> in <module>
----> 1 c = get_client_from_cli_profile(KeyClient)

~/miniconda3/envs/py37/lib/python3.7/site-packages/azure/common/client_factory.py in get_client_from_cli_profile(client_class, **kwargs)
     98         parameters['tenant_id'] = tenant_id
     99     parameters.update(kwargs)
--> 100     return _instantiate_client(client_class, **parameters)
    101
    102

~/miniconda3/envs/py37/lib/python3.7/site-packages/azure/common/client_factory.py in _instantiate_client(client_class, **kwargs)
     33         elif sys.version_info < (3, 0) and isinstance(kwargs[key], unicode):
     34             kwargs[key] = kwargs[key].encode('utf-8')
---> 35     return client_class(**kwargs)
     36
     37

TypeError: __init__() missing 2 required positional arguments: 'vault_url' and 'credential'

### PR DESCRIPTION
**Pull request by lmazuel**
Fix #7035 and #6041

```python
In [7]: from azure.common.client_factory import get_client_from_cli_profile

In [8]: from azure.keyvault.secrets import SecretClient

In [9]: client = get_client_from_cli_profile(SecretClient, vault_url="https://xxxxxxxx.vault.azure.net/")

In [10]: client.get_secret("secret").value
Out[10]: 'hush'
```
or
```python
In [1]: from azure.storage.blob import BlobServiceClient

In [2]: from azure.common.client_factory import get_client_from_cli_profile

In [3]: client = get_client_from_cli_profile(BlobServiceClient, account_url="https://xxxxxxxx.blob.core.windows.net")

In [4]: client.create_container("testclicred")
Out[4]: <azure.storage.blob._container_client.ContainerClient at 0x281ef79feb8>
```

### CODE REVIEW

File: sdk/core/azure-common/azure/common/client_factory.py [hunk 1]
```diff
@@ -22,10 +22,10 @@
 
 
 def _instantiate_client(client_class, **kwargs):
-    """Instantiate a client from kwargs, removing the subscription_id/tenant_id/base_url argument if unsupported.
+    """Instantiate a client from kwargs, removing unsupported stuff.
```
Comment (jongio) [inline sdk/core/azure-common/azure/common/client_factory.py]: I wouldn't say "stuff" in comments.  Please be more explicit

File: sdk/core/azure-common/azure/common/credentials.py [hunk 2]
```diff
@@ -28,7 +34,35 @@ def get_cli_profile():
...
+
+    def __init__(self, cli_profile):
+        if AccessToken is None:  # import failed
+            raise ImportError("You need to install 'azure-core' to use this feature")
+        self._profile = cli_profile
+
+    def get_token(self, *scopes, **kwargs):  # pylint:disable=unused-argument
+
+        if len(scopes) != 1:
+            raise ValueError("Cannot deal with multiple scope: {}".format(scopes))
```
Comment (johanste) [inline sdk/core/azure-common/azure/common/credentials.py]: Maybe rephrase to something like:

"Multiple scopes are not supported: {}"

?

File: sdk/core/azure-common/azure/common/client_factory.py (hunk 1 above)
Comment (lmazuel) [reply sdk/core/azure-common/azure/common/client_factory.py]: That's hidden method, it's why I wasn't very verbose, but yeah can make it more explciit :)

File: sdk/core/azure-common/azure/common/credentials.py (hunk 2 above)
Comment (lmazuel) [reply sdk/core/azure-common/azure/common/credentials.py]: Fixed

File: sdk/core/azure-common/azure/common/client_factory.py (hunk 1 above)
Comment (lmazuel) [reply sdk/core/azure-common/azure/common/client_factory.py]: Fixed

File: sdk/core/azure-common/azure/common/client_factory.py [hunk 3]
```diff
@@ -74,12 +74,17 @@ def get_client_from_cli_profile(client_class, **kwargs):
     """
     cloud = get_cli_active_cloud()
     parameters = {}
-    if 'credentials' not in kwargs or 'subscription_id' not in kwargs:
+    kwarg_cred = kwargs.pop('credentials', kwargs.pop('credential', None))
```
Comment (johanste) [inline sdk/core/azure-common/azure/common/client_factory.py]: If (and this needs to be double-checked) we ended up specifying an explicit `None` as credentials for storage for anonymous auth, then we probably need a sentinel value other than `None` here.

Comment (lmazuel) [reply sdk/core/azure-common/azure/common/client_factory.py]: Ok, that's fair point

Comment (lmazuel) [reply sdk/core/azure-common/azure/common/client_factory.py]: Fixed
